feature/subscriber-attributes/src/main/java/com/revenuecat/purchases/subscriberattributes/AttributionDataMigrator.kt

```diff
@@ -0,0 +1,151 @@
+package com.revenuecat.purchases.subscriberattributes
+
+import com.revenuecat.purchases.common.attribution.AttributionNetwork
+import com.revenuecat.purchases.utils.optNullableString
+import org.json.JSONObject
+
+internal object AttributionKeys {
+    const val IDFA = "rc_idfa"
+    const val IDFV = "rc_idfv"
+    const val IP = "rc_ip_address"
+    const val GPS_AD_ID = "rc_gps_adid"
+    const val NETWORK_ID = "rc_attribution_network_id"
+    internal object Adjust {
+        const val ID = "adid"
+        const val NETWORK = "network"
+        const val CAMPAIGN = "campaign"
+        const val AD_GROUP = "adgroup"
+        const val CREATIVE = "creative"
+    }
+    internal object AppsFlyer {
+        const val ID = "rc_appsflyer_id"
+        const val CAMPAIGN = "campaign"
+        const val CHANNEL = "af_channel"
+        const val MEDIA_SOURCE = "media_source"
+        const val ADSET = "adset"
+        const val AD = "af_ad"
+        const val AD_GROUP = "adgroup"
+        const val AD_KEYWORDS = "af_keywords"
+        const val AD_ID = "ad_id"
+        const val DATA_KEY = "data"
+        const val STATUS_KEY = "status"
+    }
+    internal object Branch {
+        const val CAMPAIGN = "campaign"
+        const val CHANNEL = "channel"
+    }
+    internal object MParticle {
+        const val ID = "mpid"
+    }
+}
+
+class AttributionDataMigrator {
+
+    fun convertAttributionDataToSubscriberAttributes(
+        data: JSONObject,
+        network: AttributionNetwork
+    ): Map<String, String> {
+        if (network == AttributionNetwork.APPSFLYER) {
+            val innerDataJSONObject = data.optJSONObject(AttributionKeys.AppsFlyer.DATA_KEY)
+            if (data.has(AttributionKeys.AppsFlyer.STATUS_KEY) && innerDataJSONObject != null) {
+                val keys = innerDataJSONObject.keys()
+
+                while (keys.hasNext()) {
+                    val key = keys.next()
+                    data.put(key, innerDataJSONObject[key])
+                }
+            }
+        }
```

Comment:
maybe this bit can happen inside `convertAppsFlyerAttribution`?

Reply:
True! I moved it here because one of the tests was failing, but now that I think about it the problem was with the test itself. I am adding the common attribution `rc_idfa`, `rc_idfv`, etc in that inner data json in the test, but they have to be outside. I will fix that.